pkg/sources/reconciler/awssnssource/adapter.go

```diff
@@ -47,6 +48,7 @@ var _ common.AdapterServiceBuilder = (*Reconciler)(nil)
 func (r *Reconciler) BuildAdapter(src commonv1alpha1.Reconcilable, _ *apis.URL) *servingv1.Service {
 	return common.NewMTAdapterKnService(src,
 		resource.Image(r.adapterCfg.Image),
+		resource.Label(network.VisibilityLabelKey, "public"),
```

Comment:
I'm making the `public` value up here, it doesn't actually exist.
Knative only checks whether that value differs from `cluster-local`, in which case it always falls back to public.

The reason why we need to set _some_ value is because `common.NewMTAdapterKnService()` sets this label to `cluster-local`, which we need to override.

I considered two alternative approaches to this problem:

- changing the helper to _remove_ the label when we set its value to `""` (or even adding a `RemoveLabel()` helper)
    ... but I find this hacky and unintuitive
- removing the line that sets the label to `cluster-local` by default
    ... but then we have to copy that same line to 40 targets and transformers (at the time of writing)

With this approach, only 5 components need to be changed, and setting the value to `public` makes it very explicit that the service is going to be served over a public URL, which IMO is a good thing in terms of readability.

I'm open to comments and suggestions.